components/canvas/webgl_thread.rs

```diff
@@ -1847,8 +1890,25 @@ impl WebGLImpl {
     }
 
     #[allow(unsafe_code)]
-    fn create_vertex_array(gl: &gl::Gl, chan: &WebGLSender<Option<WebGLVertexArrayId>>) {
-        let vao = gl.gen_vertex_arrays(1)[0];
+    fn create_vertex_array(
+        gl: &gl::Gl,
+        use_apple_ext: bool,
+        chan: &WebGLSender<Option<WebGLVertexArrayId>>,
+    ) {
+        let vao = if use_apple_ext {
+            match gl {
+                Gl::Gl(gl) => {
+                    let mut ids = vec![0];
+                    unsafe {
+                        gl.GenVertexArraysAPPLE(ids.len() as gl::GLsizei, ids.as_mut_ptr());
+                    }
+                    ids[0]
+                },
+                Gl::Gles(_) => unimplemented!("No GLES on macOS"),
+            }
```

Comment:
Make a thin safe wrapper around this so that the unsafe oneliner isn't repeated in multiple places.

Reply:
The unsafe code is different in each callsite. I'm not exactly sure what you are looking for.